src/style-spec/validate/validate_mapbox_api_supported.js

```diff
@@ -0,0 +1,95 @@
+// @flow
+import ValidationError from '../error/validation_error';
+
+const SUPPORTED_SPEC_VERSION = 8;
+const MAX_SOURCES_IN_STYLE = 15;
+const SOURCE_KEYS = ['type', 'url', 'tileSize'];
+
+function validateKeys(obj: Object, keys: Array<*>, errors: Array<?Error>, path: ?string): void {
+    const allowed = new Set(keys);
+    Object.keys(obj).forEach(k => {
+        if (!allowed.has(k)) {
+            const prop = path ? `${path}.${k}` : k;
+            errors.push(new ValidationError(prop, null, `Unsupported property "${prop}"`));
+        }
+    });
+}
+
+function isValidSpriteURL(url: string): boolean {
+    const validPattern = /^mapbox:\/\/sprites\/([^/]*)\/([^/]*)\/([^/]*)?$/;
+    return !!url.match(validPattern);
+}
+
+function isValidGlyphsURL(url: string): boolean {
+    const validPattern = /^mapbox:\/\/fonts\/([^/]*)\/{fontstack}\/{range}.pbf$/;
+    return !!url.match(validPattern);
+}
+
+function validateSource(sources: Object, errors: Array<?Error>): void {
+    const count = Object.keys(sources).reduce((count, key) => {
+        const source = sources[key];
+        return count + source.url.replace('mapbox://', '').split(',').length;
+    }, 0);
+
+    if (count > MAX_SOURCES_IN_STYLE) {
+        errors.push(new ValidationError('sources', null, `Styles must contain ${MAX_SOURCES_IN_STYLE} or fewer sources`));
+    }
+
+    for (const q in sources) {
+        if (!sources[q].url) {
+            errors.push(new ValidationError('sources', null, 'Source must include url'));
+        }
+
+        if (sources[q].url.indexOf('mapbox://') !== 0) {
+            errors.push(new ValidationError('sources.url', null, 'Style must reference sources hosted by Mapbox'));
+        }
+
+        validateKeys(sources[q], SOURCE_KEYS, errors, 'source');
+    }
+
+}
+
+function validate(style: Object, errors: Array<?Error>): void {
+    if (style.version !== SUPPORTED_SPEC_VERSION) {
```

Comment:
What happens to older styles - are they no longer valid, or auto updated on the backend?

Reply:
They are no longer supported by the style API.